interface-manager/src/interface/tap.rs

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use net::buffer::{PacketBuffer, PacketBufferMut};
+use net::interface::InterfaceName;
+use std::num::NonZero;
+use std::os::fd::AsRawFd;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tracing::{debug, error, info};
+
+#[derive(Debug)]
+#[repr(transparent)]
+pub struct TapDevice {
+    file: tokio::fs::File,
+}
+
+mod helper {
+    /// This is a validated type around a value which is regrettably fragile.
+    ///
+    /// 1. Passed directly to the kernel.
+    /// 2. By a privileged thread.
+    /// 3. In an ioctl.
+    /// 4. By an implicitly null terminated pointer.
+    ///
+    /// As a result, strict checks are in place to ensure memory integrity.
+    ///
+    /// <div class=warning>
+    ///
+    /// It is essential that this type remains transparent.
+    /// Only zero-sized types may be added to this structure as we don't control the ABI.
+    /// We are subject to a contract with the kernel.
+    /// </div>
+    #[repr(transparent)]
+    #[derive(Debug, Copy, Clone)]
+    pub(super) struct InterfaceRequest(libc::ifreq);
+
+    use net::interface::InterfaceName;
+    use nix::libc;
+
+    nix::ioctl_write_ptr_bad!(
+        /** Create a tap device */
```

Comment:
Oh no, Fredi's C-style comments are contagious! :scream_cat:

Reply:
They are valid rust comments! https://doc.rust-lang.org/reference/comments.html

Reply:
Yes they are!

I just prefer sticking to a single type, for consistency, that's all. And I've a small preference for the double-slash `//`, which doesn't prevent you from surrounding blocks with comments with additional `/* */` markers to temporarily exclude something from compilation. But that's fine! :slightly_smiling_face:

Reply:
This is actually more subtle than you might think.  `/**` and `///` are rustdoc, not just comments.  Technically they are a type of attribute.  In this case I am forcing nix to generate a doc string from the ioctl wrapper it cooks